suite-native/module-trading/src/utils/general/utils.ts

```diff
@@ -208,3 +213,17 @@ export const getTradeTitle = (trade: TradingTransaction, translate: Translate) =
 
 export const getFormDraftKeyPrefixFromTradingType = (tradingType: TradingType) =>
     `trading-${tradingType}` as const satisfies FormDraftKeyPrefix;
+
+export const toCaseAwareCryptoId = (cryptoId: CryptoId): CryptoId => {
+    const { network, contractAddress } = cryptoIdToNetworkAndContractAddress(cryptoId);
```

Comment:
I think we want to use network `tradeCryptoId` instead of `coingeckoId`.

Or do something like:

```suggestion

const { symbol, contractAddress } = cryptoIdToNetworkSymbolAndContractAddress(cryptoId);
if (!contractAddress || (contractAddress && isCryptoIdForNativeToken(cryptoId))) {   
	return cryptoId;
}
const adjustedContractAddress = getContractAddressForNetworkSymbol(symbol, contractAddress);

return toTokenCryptoId(symbol, adjustedContractAddress);
```

Reply:
this also doesnt crash app with invariant in case network is not found

Reply:
Updated in b0c74deee5fb8e6423ab3772e62ef5ee6eaa151c.